src/libaktualizr/primary/commands.cc

```diff
@@ -4,11 +4,17 @@
 
 namespace command {
 
-Json::Value BaseCommand::toJson() {
-  Json::Value json;
+std::string BaseCommand::toJson() {
+  Json::Value json = Json::Value();
+  return BaseCommand::toJson(json);
+}
+
+std::string BaseCommand::toJson(Json::Value json) {
   json["variant"] = variant;
-  json["fields"] = Json::Value(Json::arrayValue);
-  return json;
+  if (!json.isMember("fields")) {
+    json["fields"] = Json::Value(Json::arrayValue);
+  }
+  return Json::FastWriter().write(json);
 }
 
 std::shared_ptr<BaseCommand> BaseCommand::fromPicoJson(const picojson::value& json) {
```

Comment:
Can wait but it would be nice to remove picojson and only use one library. Now there is this weird mix in `commands.cc`. @OYTIS

Reply:
Unfortunatelly  we cannot remove it because it supports stream parsing which is needed for socket gateways.

Reply:
But do we really need this socket gateway functionality?